packages/jaeger-ui/src/components/TracePage/TracePageHeader/SpanGraph/index.tsx

```diff
@@ -31,57 +32,38 @@ type SpanGraphProps = {
   updateNextViewRangeTime: (nextUpdate: ViewRangeTimeUpdate) => void;
 };
 
-/**
- * Store `items` in state so they are not regenerated every render. Otherwise,
- * the canvas graph will re-render itself every time.
- */
-type SpanGraphState = {
-  items: {
-    valueOffset: number;
-    valueWidth: number;
-    serviceName: string;
-  }[];
+type SpanItem = {
+  valueOffset: number;
+  valueWidth: number;
+  serviceName: string;
 };
 
-function getItem(span: Span) {
+function getItem(span: Span): SpanItem {
   return {
     valueOffset: span.relativeStartTime,
     valueWidth: span.duration,
     serviceName: span.process.serviceName,
   };
 }
 
-export default class SpanGraph extends React.PureComponent<SpanGraphProps, SpanGraphState> {
-  state: SpanGraphState;
+function getItems(trace: Trace): SpanItem[] {
+  return trace.spans.map(getItem);
+}
+
+const memoizedGetItems = memoizeOne(getItems);
 
+export default class SpanGraph extends React.PureComponent<SpanGraphProps> {
```

Comment:
> React.PureComponent<SpanGraphProps>

Is`SpanGraphProps` still relevant? Seems like it was only affecting constructor/callback.

Reply:
ah, I guess it affects `this.props` in render()